zubhub_frontend/zubhub/src/components/upload_file/video.jsx

```diff
@@ -0,0 +1,248 @@
+import React, {useState} from 'react'
+import VideoIcon from '@material-ui/icons/Movie';
+import CheckIcon from '@material-ui/icons/Check';
+import CloseIcon from '@material-ui/icons/Close';
+import MovieIcon from '@material-ui/icons/Movie';
+import InsertLinkIcon from '@material-ui/icons/InsertLink';
+import {
+  Grid,
+  Box,
+  Container,
+  Card,
+  CardActionArea,
+  CardContent,
+  Chip,
+  Dialog,
+  Select,
+  MenuItem,
+  Typography,
+  CircularProgress,
+  OutlinedInput,
+  FormHelperText,
+  FormControl,
+  InputLabel,
+  ClickAwayListener,
+} from '@material-ui/core';
+
+function VideoToggle(props) {
+  const [state, setState] = React.useState({
+    ...JSON.parse(JSON.stringify(vars.default_state)),
+  });  
```

Comment:
please use cloneDeep instead and please do this inside a callback, so it doesn't do the expensive operations on every render
```suggestion
  const [state, setState] = React.useState(() => cloneDeep(vars.default_state));  
```